### ISSUE TITLE
Attempting to upgrade the global blitz version from the blitz new prompt fails

### ISSUE THREAD

**Issue by Roesh** [blitz-js/blitz#1998]
**What is the problem?**

EDIT: There is a really simple workaround mentioned by Brandon in the posts below, npm i -g blitz@latest will upgrade the global blitz version to the latest release.

When trying to create a new blitzjs project, if your global blitz version is not the latest released version, a prompt to upgrade the global blitz version appears. Selecting yes to the prompt causes the following error:
```
blitz new linuxJs
You are using beta software - if you have any problems, please open an issue here:
      https://github.com/blitz-js/blitz/issues/new/choose

✔ Your global blitz version is outdated. Upgrade? · yes
    TypeError: Cannot read property 'toString' of null
```

**What are detailed steps to reproduce this?**

1. Create a new blitz app with blitz new <app name> when your global version is not the latest version
2. Accept the prompt to update the global blitz version
3. The error should occur

**Run `blitz -v` and paste the output here:**

```
Linux 5.4 | linux-x64 | Node: v14.15.5

blitz: 0.30.0-canary.13 (global)
blitz: 0.30.0-canary.13 (local)

  Package manager: npm 
  System:
    OS: Linux 5.4 Linux Mint 19.3 (Tricia)
    CPU: (16) x64 AMD Ryzen 7 2700 Eight-Core Processor
    Memory: 3.97 GB / 15.63 GB
    Shell: 4.4.20 - /bin/bash
  Binaries:
    Node: 14.15.5 - /usr/local/bin/node
    Yarn: Not Found
    npm: 7.5.4 - /usr/local/bin/npm
    Watchman: Not Found
  npmPackages:
    @prisma/client: 2.17.0 => 2.17.0 
    blitz: 0.30.0-canary.13 => 0.30.0-canary.13 
    prisma: 2.17.0 => 2.17.0 
    react: 0.0.0-experimental-3310209d0 => 0.0.0-experimental-3310209d0 
    react-dom: 0.0.0-experimental-3310209d0 => 0.0.0-experimental-3310209d0 
    typescript: 4.1.5 => 4.1.5

```

**Comment by flybayer** [comment]
@Roesh can you please update to 0.30.3 and try again?

**Comment by Roesh** [comment]
Sounds good. I'll need to learn a little more about how npx works since I am on 0.30.0-canary.15 and I'm not sure what the best method to create a single project with 0.30.3. I should be able to have that done by sometime tomorrow

**Comment by flybayer** [comment]
@roesh run `npm i -g blitz@latest` and then run `blitz new` to create a new project with latest version

**Comment by Roesh** [comment]
Cool. So I upgraded to blitz 0.30.3 with npm i -g blitz@0.30.3, and I still run into the same issue. 

If I upgrade to latest (0.30.4), I don't get the `✔ Your global blitz version is outdated. Upgrade?` prompt. Is there a way to manually trigger that? I can wait till we have 0.30.5 to test latest

**Comment by flybayer** [comment]
So maybe there's some confusion here 🤔 `✔ Your global blitz version is outdated. Upgrade?` only upgrades the globally installed CLI version. That doesn't affect your app code.

The global CLI version only affects what is generated via `blitz new`. Everything else is determined by your local blitz version. When you use global blitz cli inside a project, it uses the cli code from your project instead of the global install

**Comment by Roesh** [comment]
Hmm that makes sense, let me make sure I'm doing this right

Opening VSCode to a folder that is not a blitz project: 

```
~/Documents/DeleteMe$ ls
linuxJsFiles  starter

```

```
~/Documents/DeleteMe$ blitz -v
You are using beta software - if you have any problems, please open an issue here:
      https://github.com/blitz-js/blitz/issues/new/choose

Linux 5.4 | linux-x64 | Node: v14.15.5

blitz: 0.30.4 (global)

  Package manager: npm 
  System:
    OS: Linux 5.4 Linux Mint 19.3 (Tricia)
    CPU: (16) x64 AMD Ryzen 7 2700 Eight-Core Processor
    Memory: 8.11 GB / 15.63 GB
    Shell: 4.4.20 - /bin/bash
  Binaries:
    Node: 14.15.5 - /usr/local/bin/node
    Yarn: Not Found
    npm: 7.5.4 - /usr/local/bin/npm
    Watchman: Not Found
  npmPackages:
    @prisma/client: Not Found
    blitz: Not Found
    prisma: Not Found
    react: Not Found
    react-dom: Not Found
    typescript: Not Found
```

```
~/Documents/DeleteMe$ blitz new testApp
You are using beta software - if you have any problems, please open an issue here:
      https://github.com/blitz-js/blitz/issues/new/choose

✔ Your global blitz version is outdated. Upgrade? · yes   <--------------------- I select yes here
    TypeError: Cannot read property 'toString' of null  <------------------ I get this error
```

I assume that since I am not in a blitz project that the global blitz version would be used when generating the project correct? If I upgrade to the latest global repository (which currently is 0.30.5), I won't get the prompt to upgrade blitz versions until there's a new global version (0.30.6)

**Comment by Roesh** [comment]
I finally read through the contributors guide, I've been meaning to get around to it for too long, looks like its pretty simple, I just had to fork and clone the repo.

There are two spots that might be erroring, they're in packages/cli/src/commands/new.ts, lines 88 or line 97 are causing the error

`const useYarn = checkYarn.stdout.toString().includes("blitz@")`

`versionResult.stdout.toString().match(/(?<=blitz: )(.*)(?= \(global\))/) || []`

I'm working my way through testing using the cloned repo currently and hope to have some progress by tomorrow

**Comment by flybayer** [comment]
Awesome, thanks @Roesh!

**Comment by Roesh** [comment]
Anytime! 

So, when I was trying to run blitz new in the auth example I ran into an issue because I needed yarn dev to be running for it to work. I didn't have yarn installed, so I downloaded yarn via npm and did yarn dev. I modified new.ts to always ask to upgrade, and ran blitz new and I didn't get the error, I'm assuming its because I actually had yarn installed now lolol,

I went to a random directory and ran blitz new and the upgrade prompt works fine (Asks to upgrade 0.30.4 to 0.30.5 and , so installing yarn to run the development blitz server fixed this issue.

Do we expect all users to download yarn in addition to npm, or should we do a null check to make sure checkYarn actually has a non null buffer for stdout? So like 
```if(checkYarn !== null && checkYarn.stdout !== null){
            const useYarn = checkYarn.stdout.toString().includes("blitz@")
            const upgradeOpts = useYarn ? ["global", "add", "blitz"] : ["i", "-g", "blitz@latest"]

            spawn.sync(useYarn ? "yarn" : "npm", upgradeOpts, {stdio: "inherit"})
          }```?
